js/src/checkout-drop-in.js

```diff
@@ -1,246 +1,23 @@
-	 */
-	const handle_error = error => {
-		// Check syntax error name.
-		if ( 'SyntaxError' === error.name ) {
-			error.message = pronamicPayAdyenCheckout.syntaxError;
-		}
-
-		// Show error message.
-		dropin.setStatus( 'error', { message: error.message } );
-	}
-
-	/**
-	 * Get payment methods configuration.
-	 *
-	 * @return object
-	 */
-	const getPaymentMethodsConfiguration = () => {
-		/**
-		 * Complement Apple Pay configuration with `onValidateMerchant`
-		 * callback when using your own Apple Pay certificate.
-		 * 
-		 * @link https://github.com/pronamic/wp-pronamic-pay-adyen/issues/5#issue-1154083692
-		 */
-		if ( pronamicPayAdyenCheckout.paymentMethodsConfiguration.applepay && pronamicPayAdyenCheckout.applePayMerchantValidationUrl ) {
-			pronamicPayAdyenCheckout.paymentMethodsConfiguration.applepay.onValidateMerchant = ( resolve, reject, validationUrl ) => {
-				send_request(
-					pronamicPayAdyenCheckout.applePayMerchantValidationUrl,
-					{
-						validation_url: validationUrl
-					}
-				)
-				.then( validate_response )
-				.then( data => {
-					// Handle Apple error.
-					if ( data.statusMessage ) {
-						throw new Error( data.statusMessage, {
-							cause: data
-						} );
-					}
-
-					resolve( data );
-				} )
-				.catch( error => {
-					handle_error( error );
-
-					// Reject to dismiss Apple Pay overlay.
-					reject( error );
-				} );
-			};
-		}
-
-		// Compliment PayPal configuration.
-		if ( pronamicPayAdyenCheckout.paymentMethodsConfiguration.paypal ) {
-			pronamicPayAdyenCheckout.paymentMethodsConfiguration.paypal.onCancel = ( data, dropin ) => {
-				dropin.setStatus( 'ready' );
-			};
-		}
+'use strict';
 
-		return pronamicPayAdyenCheckout.paymentMethodsConfiguration;
-	};
+( async function () {
 
 	/**
 	 * Adyen Checkout.
 	 */
-	const checkout = new AdyenCheckout( pronamicPayAdyenCheckout.configuration );
-
-	const dropin = checkout.create( 'dropin', {
-		paymentMethodsConfiguration: getPaymentMethodsConfiguration(),
-		onSelect: dropin => {
-			let configuration = pronamicPayAdyenCheckout.configuration;
-
-			if ( false === configuration.showPayButton ) {
-				dropin.submit();
-			}
+	const configuration = {
+		...pronamicPayAdyenCheckout.configuration,
+		onPaymentCompleted: ( result, component ) => {
+			window.location.href = pronamicPayAdyenCheckout.paymentReturnUrl;
 		},
-		onSubmit: ( state ) => {
-			// Set loading status to prevent duplicate submits.
-			dropin.setStatus( 'loading' );
-
-			send_request(
-				pronamicPayAdyenCheckout.paymentsUrl,
-				state.data
-			)
-			.then( validate_response )
-			.then( process_response )
-			.catch( handle_error );
-		},
-		onAdditionalDetails: ( state ) => {
-			send_request(
-				pronamicPayAdyenCheckout.paymentsDetailsUrl,
-				state.data
-			)
-			.then( validate_response )
-			.then( process_response )
-			.catch( handle_error );
+		onError: ( error, component ) => {
+			console.error( error.name, error.message, error.stack, component );
```

Comment:
Update drop-in status instead of logging to console?

```suggestion
			dropinComponent.setStatus( 'error', { message: error.message } );
```